api/include/opentelemetry/plugin/detail/utility.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include "opentelemetry/nostd/string_view.h"
+
+namespace opentelemetry
+{
+namespace plugin
+{
+namespace detail
+{
+inline void CopyErrorMessage(const char *source, std::string &destination) noexcept
```

Comment:
Since this isn't a virtual, can it just use `std::string`?

Reply:
It's using a std::string for the destination.

It takes a `const char*` instead of string_view or string for the source to handle the case when it might be a nullptr.